auth/app/Http/Controllers/AuthController.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Cache;
+use Laravel\Lumen\Routing\Controller as BaseController;
+use Lcobucci\JWT\Builder;
+use Lcobucci\JWT\Parser;
+use Lcobucci\JWT\Signer\Hmac\Sha256;
+use Miner\Auth\Entities\User;
+use Miner\Auth\Repositories\UserRepository;
+use Miner\Auth\Entities\Uuid;
+
+class AuthController extends BaseController
+{
+    public function login(Request $request, UserRepository $userRepository, Cache $cache)
+    {
+        $this->validate($request, [
+            'username' => 'required',
+            'password' => 'required'
+        ]);
+
+        $email = $request->get('email');
+        $password = $request->get('password');
+
+        $user = $userRepository->findByEmail($email);
+
+        // use php 5.5 password_hash/password_verify magix to validate the password matches
+        if(!password_verify($password, $user->getPasswordHash())) {
+            throw new \Exception('invalid password');
+        }
+
+        $accessToken = (string)$this->generateAccessToken($user);
+
+        // generate a new unique id for our token
+        $uuid = new Uuid();
+
+        // get all refresh tokens for this user based on the
+        $refreshTokens = json_decode($cache->get($user->getCacheKey()));
+        $refreshTokens[] = $uuid->toString();
+
+        // get rid of the oldest token to make room for the new token
+        if(count($refreshTokens) >= 10) {
+            array_shift($refreshTokens);
+        }
+
+        // add the new token to cache
+        $cache->set($user->getCacheKey(), json_encode($refreshTokens));
+
+        return response()->json(['accessToken' => $accessToken, 'refreshToken' => $uuid->toString()]);
+    }
+
+    public function refresh(Request $request, UserRepository $userRepository, Cache $cache)
+    {
+        $this->validate($request, [
+            'accessToken' => 'required',
+            'refreshToken' => 'required'
+        ]);
```

Comment:
Same as above, prefer validation rules as properties or another class that is referenced.